lib/puppet/http/external_client.rb

```diff
@@ -57,8 +59,6 @@ def post(url, headers: {}, params: {}, options: {}, &block)
       options[:basic_auth] = { user: options[:user], password: options[:password] }
     end
 
-    body = options.fetch(:body) { |_| raise ArgumentError, "'post' requires a 'body' option" }
-    headers['Content-Type'] = options[:content_type]
```

Comment:
should we add an error here if `content-type` isn't included in the header hash?

Reply:
Mostly because I'm not sure if puppetserver's http client requires that, for example if the request body is empty? So I figured it'd better to just pass what we have to the external client and let it deal with it (like it does already).

Reply:
oooh, that makes a ton of sense to do it that way